### PR TITLE
Bugfix/make graffiti valid ida plugin

### PR DESCRIPTION
**Pull request by urielmalin**
basic refactor graffiti into IDA expected plugin format
please merge after https://github.com/yoavst/Graffiti/pull/6

### CODE REVIEW

File: backends/ida/graffiti.py [hunk 1]
```diff
@@ -370,6 +306,100 @@ def renamed(self, ea, new_name, is_local):
             }
             lengthy_send(sock, to_bytes(json.dumps(payload)))
 
-hooks = [GraffitiUIHooks(), GrafitiIDBHooks()]
-for hook in hooks:
-    hook.hook()
+
+class GraffitiPlugin(idaapi.plugin_t, idaapi.UI_Hooks):
+    flags = idaapi.PLUGIN_MOD | idaapi.PLUGIN_HIDE
+    comment = "Add support for CPP navigating on top of ida_kernelcache"
```
Comment (yoavst) [inline backends/ida/graffiti.py:312]: Wrong comment :)
```suggestion
    comment = "Create customized callgraph directly from IDA"
```

File: backends/ida/graffiti.py [hunk 2]
```diff
@@ -370,6 +306,100 @@ def renamed(self, ea, new_name, is_local):
...
+    help = ""
+    wanted_name = "Graffiti"
+    wanted_hotkey = ""
+
+    def __init__(self):
+        super().__init__()
```
Comment (yoavst) [inline backends/ida/graffiti.py:318]: idapython2 compatibility:

```suggestion
        super(idaapi.plugin_t, self).__init__()
```

**Review by yoavst — CHANGES_REQUESTED** [review]
Can you also modify [the docs](https://github.com/yoavst/Graffiti/blob/master/docs/platforms/IDA.md) with instructions how to install ida plugin?

You can add "installing into user directory" section from [here](https://hex-rays.com/blog/igors-tip-of-the-week-103-sharing-plugins-between-ida-installs/)

(Also apply comments from #6

File: backends/ida/graffiti.py (hunk 1 above)
Comment (urielmalin) [reply backends/ida/graffiti.py:312]: haha

**Comment by urielmalin** [comment]
Will update docs later, for now just put it here for a reference:

> clone the repository and symlink  ~/.idapro/plugins/ida_kcpp_plugin.py to ida_kcpp_plugin.py

**Pull request merged**